src/platform/plugins/shared/controls/public/types.ts

```diff
@@ -25,6 +25,14 @@ export const isClearableControl = (control: unknown): control is CanClearSelecti
   );
 };
 
+export interface CanClearVariables {
```

Comment:
How about moving `CanClearVariables` and is `isVariablesControl` types to src/platform/plugins/shared/controls/public/controls/esql_control/types.ts since the type only exists on ESQL controls?

Reply:
Ofc https://github.com/elastic/kibana/pull/202875/commits/0611ecf30299910457467360e3ae0a12d6c93c7e